dashboard/app/models/reference_guide.rb

```diff
@@ -0,0 +1,70 @@
+require 'fileutils'
+
+# == Schema Information
+#
+# Table name: reference_guides
+#
+#  id                :bigint           not null, primary key
+#  name              :string(255)
+#  slug              :string(255)
+#  course_version_id :integer
+#  content           :text(65535)
+#  order             :integer
+#  created_at        :datetime         not null
+#  updated_at        :datetime         not null
+#
+class ReferenceGuide < ApplicationRecord
+  belongs_to :course_version
+
+  validates_uniqueness_of :slug, scope: :course_version_id
+
+  def serialize
+    {
+      id: id,
+      name: name,
+      slug: slug,
+      course_version_id: course_version_id,
+      content: content,
+      order: order
+    }
+  end
+
+  def write_serialization
+    return unless Rails.application.config.levelbuilder_mode
+    file_path = Rails.root.join("config/reference_guides/#{slug}.json")
```

Comment:
Is slug is only unique within the course version does this file path need to include the course version so you don't end up with two files in different course versions with the same slug causing issues?

Reply:
I think technically right now you could have a unit_group content root with the same name as a unit content root so I think the best bet would be to lean into course_offering.key + course_version.key